src/pages/messages/components/NviCandidateActions.tsx

```diff
@@ -233,30 +243,94 @@ export const NviCandidateActions = ({ nviCandidate, nviCandidateQueryKey }: NviC
               {t('tasks.nvi.reject_nvi_candidate')}
             </Button>
 
-            <Divider sx={{ mb: '1rem' }} />
+            <RejectionDialog
+              open={hasSelectedRejectCandidate}
+              onCancel={() => setHasSelectedRejectCandidate(false)}
+              onAccept={async (reason) => {
+                await statusMutation.mutateAsync({ status: 'Rejected', reason });
+                setHasSelectedRejectCandidate(false);
+              }}
+              isLoading={statusMutation.isPending}
+            />
 
-            {hasSelectedRejectCandidate && (
-              <MessageForm
-                confirmAction={async (reason) => {
-                  await statusMutation.mutateAsync({ status: 'Rejected', reason });
-                  setHasSelectedRejectCandidate(false);
-                }}
-                cancelAction={() => setHasSelectedRejectCandidate(false)}
-                fieldLabel={t('tasks.nvi.reject_nvi_candidate_form_label')}
-                buttonTitle={t('tasks.nvi.reject_nvi_candidate')}
-              />
-            )}
+            <Divider sx={{ mb: '1rem' }} />
           </>
         )}
 
-        {!hasSelectedRejectCandidate && (
-          <MessageForm
-            confirmAction={async (text) => await createNoteMutation.mutateAsync({ text })}
-            fieldLabel={t('tasks.nvi.note')}
-            buttonTitle={t('tasks.nvi.save_note')}
-          />
-        )}
+        <MessageForm
+          confirmAction={async (text) => await createNoteMutation.mutateAsync({ text })}
+          fieldLabel={t('tasks.nvi.note')}
+          buttonTitle={t('tasks.nvi.save_note')}
+        />
       </Box>
     </>
   );
 };
+
+interface RejectionDialogProps {
+  open: boolean;
+  onCancel: () => void;
+  onAccept: (reason: string) => Promise<unknown> | void;
+  isLoading: boolean;
+}
+
+const maxLength = 160;
+
+const RejectionDialog = ({ open, onCancel, onAccept, isLoading }: RejectionDialogProps) => {
+  const { t } = useTranslation();
+  const [reason, setReason] = useState('');
+
+  const handleAccept = async () => {
+    await onAccept(reason.trim());
+    setReason('');
+  };
+
+  const handleClose = () => {
+    onCancel();
+    setReason('');
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setReason(event.target.value);
+  };
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>{t('tasks.nvi.reject_nvi_candidate')}</DialogTitle>
+      <DialogContent>
+        <Typography gutterBottom>{t('tasks.nvi.reject_nvi_candidate_modal_text')}</Typography>
+        <form onSubmit={handleAccept}>
+          <TextField
+            value={reason}
+            onChange={handleChange}
```

Comment:
FYI: personlig liker jeg å ha sånne små metoder inline, siden man  da automatisk får riktige typer på `event`.
Sjelden særlig stor praktisk betydning, men siden vi her har mulitiline og det blir `<textarea>` i HTML, mener jeg typen her egentlig vil være `ChangeEvent<HTMLTextAreaElement>` og ikke `ChangeEvent<HTMLInputElement>` som er angitt i `handleChange`. Dermed kan man miste litt presisjon og ende opp med feil type.

```suggestion
            onChange={(event) => setReason(event.target.value)}
```

Reply:
Det kan forsåvidt være et argument for å droppe egen state til å håndtere verdiene også da, ref https://github.com/BIBSYSDEV/NVA-Frontend/pull/5602/files#r1598547298. Siden ved å bruke det som native gjøres ved en HTML form bare vil returnere stregnen/verdien som er i feltet, uten at vi trenger å gjøre noe jobb.
Vi må bare passe på at det blir riktig `name` på feltet